main.go

```diff
@@ -62,7 +62,6 @@ Common Options:
 // usage will print out the flag options for the server.
 func usage() {
 	fmt.Printf("%s\n", usageStr)
-	os.Exit(0)
```

Comment:
I will update the PR, but we call `flag.Parse()` which is `CommandLine.Parse(os.Args[1:])`. and `CommandLine` is defined as `var CommandLine = NewFlagSet(os.Args[0], ExitOnError)`. Which means that if there is a parsing error, `flag` will call `os.Exit(2)`. Notice that it does not return `0`.
Two ways of fixing:
1- I keep `usage()` as it is now and do `flag.Usage = func() { fmt.Printf("%s\n", usageStr) }
2- I silence this staticcheck warning
Which one do you prefer?